src/hooks/signer/useSendTx.ts

```diff
@@ -0,0 +1,214 @@
+import { ApiPromise, SubmittableResult } from '@polkadot/api';
+import { SignerOptions } from '@polkadot/api/submittable/types';
+import { SignerResult, SubmittableExtrinsic } from '@polkadot/api/types';
+import type { DefinitionRpcExt } from '@polkadot/types/types';
+import { KeyringPair } from '@polkadot/keyring/types';
+import {
+  QueueTx,
+  QueueTxMessageSetStatus,
+  QueueTxStatus,
+  QueueTxResult,
+} from '@/types/Status';
+import type { Option } from '@polkadot/types';
+import type { Multisig, Timepoint } from '@polkadot/types/interfaces';
+import type { AddressProxy } from '@/types/Signer';
+import { keyring } from '@polkadot/ui-keyring';
+import { web3FromSource } from '@polkadot/extension-dapp';
+import AccountSigner from './AccountSigner';
+import { useApi } from '@/hooks';
+import BN from 'bn.js';
+import { cacheUnlock, extractExternal, handleTxResults } from './util';
+
+export default function useSendTx() {
+  async function submitRpc(
+    api: ApiPromise,
```

Comment:
yeah, that makes sense. but the reason why I pass `api: ApiPromise` instead of using useApi is that `api` goes undefined when I using useApi func sometimes. it seems not working well. maybe should refactor all codes regarding `polkadotContext` related files.